TourMate/TourMate/Frontend/Plan/Views/PlanViews/PlanDisplayView.swift

```diff
@@ -10,8 +10,8 @@ import SwiftUI
 struct PlanDisplayView<T: Plan, Content: View>: View {
 
     @ObservedObject var planDisplayViewModel: PlanDisplayViewModel<T>
-    let commentsViewModel: CommentsViewModel
-    let planUpvoteViewModel: PlanUpvoteViewModel
+    @ObservedObject var commentsViewModel: CommentsViewModel
+    @ObservedObject var planUpvoteViewModel: PlanUpvoteViewModel
```

Comment:
was it wrong previously?

Reply:
I'm not very sure also but I don't think so? Previously these VMs don't need to be stateful/observed in the views until they reach the components when they're set to @StateObject. But now multiple components are sharing the views so I shift the `@StateObject` to the parent view.

Reply:
Ah I see